ee/tabby-webserver/src/service/repository.rs

```diff
@@ -25,18 +37,57 @@ impl RepositoryService for DbConn {
     }
 
     async fn create_repository(&self, name: String, git_url: String) -> Result<ID> {
-        Ok(self.create_repository(name, git_url).await?.as_id())
+        Ok((self as &DbConn)
+            .create_repository(name, git_url)
+            .await?
+            .as_id())
     }
 
     async fn delete_repository(&self, id: &ID) -> Result<bool> {
-        Ok(self.delete_repository(id.as_rowid()?).await?)
+        Ok((self as &DbConn).delete_repository(id.as_rowid()?).await?)
     }
 
     async fn update_repository(&self, id: &ID, name: String, git_url: String) -> Result<bool> {
-        self.update_repository(id.as_rowid()?, name, git_url)
+        (self as &DbConn)
+            .update_repository(id.as_rowid()?, name, git_url)
             .await?;
         Ok(true)
     }
+
+    async fn search_files(
+        &self,
+        name: String,
+        path_glob: String,
+        top_n: usize,
+    ) -> Result<Vec<FileEntry>> {
+        let git_url = self.get_repository_git_url(name).await?;
+        let matching = SourceFile::all()
+            .map_err(anyhow::Error::from)?
+            .filter_map(|file| {
+                if file.git_url == git_url && glob_matches(&path_glob, &file.filepath) {
+                    Some(FileEntry {
+                        r#type: "file".into(), // Directories are not currently stored in files.jsonl
+                        path: file.filepath,
+                    })
+                } else {
+                    None
+                }
+            })
+            .take(top_n)
+            .collect();
+        Ok(matching)
+    }
+
+    async fn repository_meta(&self, name: String, path: String) -> Result<RepositoryMeta> {
```

Comment:
Please refactor https://github.com/TabbyML/tabby/blob/07f13f5ce94f03ad9b4207a96af970d64d9d3726/ee/tabby-webserver/src/repositories/resolve.rs#L28 to make it shared between this service and resolve.rs